[Max Tan] iP

maxtance opened the pull request:
# AnyaPro
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is,
1. download it from [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE!**

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
#
If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

maxtance committed:
Added Anya.java. Can Greet, Echo, and Exit.

maxtance committed:
Level-2: Added add and list functions to Anya.

maxtance committed:
Level 3: Added Task.java. Implemented marking and unmarking a task.

maxtance committed:
Added Todo.java, Event.java, Deadline.java.

maxtance committed:
Added automated testing Anya.java

maxtance committed:
Added AnyaException file. Included some error handling - ensure valid commands.

maxtance committed:
Added delete feature. Updated automated test case.

maxtance committed:
Level-7: Add save and load data functionality to Anya.java

maxtance committed:
Level-8: Update Deadline tasks to take in LocalDateTime as parameter

maxtance committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Anya.java

maxtance committed:
A-MoreOOP: Add TaskList class

maxtance committed:
A-MoreOOP: Add Ui file and restructure Anya file

maxtance committed:
A-MoreOOP: Add Storage class that deals with loading and saving tasks into a file

maxtance committed:
A-MoreOOP: Add Parser class that breaks down user input to commands

maxtance committed:
A-Packages: Create Anya and Anya.task package

maxtance committed:
Merge remote-tracking branch 'upstream/add-gradle-support'
Fetch Gradle support to ongoing project from upstream

maxtance committed:
Set up gradle to include checkstyle

maxtance committed:
A-JUnit: Add JUnit tests for Parser and TaskList

maxtance committed:
Update mainClassName in build.gradle

maxtance committed:
Add header comments to some non-private methods

maxtance committed:
A-CodingStandard: Update code to comply with coding standard

maxtance committed:
Level-9: Add Find feature

maxtance committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/anya/task/Task.java

maxtance committed:
Merge branch 'branch-A-CodingStandard'

maxtance committed:
Update code to comply with coding standard

maxtance committed:
Fix coding standards

maxtance committed:
Add javaFX dependencies to build.gradle

njxue commented on src/main/java/anya/Anya.java:
```diff
@@ -0,0 +1,187 @@
+        } finally {
+            this.ui.loadFileSuccessMessage();
+            list(tasks);
+        }
```
Great that you added a `finally` block 👍 . I didn't think of that in my own implementation!

njxue commented on src/main/java/anya/AnyaException.java:
```diff
@@ -0,0 +1,12 @@
+    public String getMessage() {
+        return super.getMessage();
+    }
+}
```
Would it be better to create child classes that extend `AnyaException`? That way, when you use the `throw` or `throws` keyword in your code, the type of exception being thrown is more explicit (eg. `UnknownAnyaCommandException`).

njxue commented on src/main/java/anya/Storage.java:
```diff
@@ -0,0 +1,80 @@
+        File savedTask = new File(this.filePath);
+        if (!savedTask.exists()) {
+            savedTask.createNewFile();
+        }
```
Nice use of Guard Clauses!

njxue commented on src/main/java/anya/Ui.java:
```diff
@@ -0,0 +1,111 @@
+        }
+        breakLine();
+    }
+}
```
Would it be better to prepend some of the methods with `get` (eg. `getLoadingFileMessage()`)? Because at first glance, the job of the `loadingFileMessage()` method may look like it is _loading_ the message instead of _retrieving_ the message.

njxue reviewed (commented):
Overall very neat code 😮 , I personally learnt some stuff from reading through your code!

maxtance committed:
Add a skeleton for GUI.

izzahaj commented on src/main/java/anya/Ui.java:
```diff
@@ -0,0 +1,111 @@
+        breakLine();
+    }
+
+    public void deleteTaskMessage(Task task) {
```
Perhaps a clearer method name here? E.g. `showDeleteTaskMessage`. I noticed the same issue in several other methods too.

izzahaj commented on src/main/java/anya/Anya.java:
```diff
@@ -0,0 +1,187 @@
+        TaskList filteredTasks = tasks.getMatchingTasks(keyword);
+        this.ui.filteredTaskMessage(filteredTasks, keyword);
+    }
+}
```
Would it be better to separate the methods handling the commands into a few Command classes?

izzahaj commented on src/main/java/anya/Anya.java:
```diff
@@ -0,0 +1,187 @@
+                } else {
+                    throw new AnyaException("Anya doesn't understand this command.");
+                }
+
```
Any reason why you chose to use if-else statements over switch statements?

izzahaj reviewed (commented):
In general, I found your code quite readable except the main Anya class which is seems too long. Apart from the naming of some of the methods, I think you did a good job adhering to the coding standards.

maxtance committed:
Update GUI to run commands implemented at Level 9.

maxtance committed:
Fix styling erros.

maxtance committed:
Remove uncessary methods and imports

maxtance committed:
Add assertions to check internal invariants and postconditions

Assertions are added to test for assumptions about how the code should
behave during runtime. This is important as some assumptions might not
hold true and can help to test for potential bugs in the code.

maxtance committed:
Merge pull request #1 from maxtance/branch-A-Assertions

First pull request from branch-A-Assertions to Master branch

maxtance committed:
Improve code quality.

Made the code obvious by using enumerations and switch-case instead of
comparing equality of strings in if-else statements. Included default
case for switch-case.

Deleted code that are not in use. Specifically text-ui-test since a GUI
is being used instead.

maxtance committed:
Merge pull request #2 from maxtance/branch-A-CodeQuality

Improve code quality.

maxtance committed:
Set up Continuous Integration

maxtance committed:
Add feature to show statistics and to create a new task list

maxtance committed:
Add Javadocs for some methods and update code to comply with coding

maxtance committed:
Add styling to JavaFx

maxtance committed:
UI improvement to javaFx.

maxtance committed:
Add product screen shot

maxtance committed:
Fix Ui bug

maxtance committed:
Update Ui.png

maxtance committed:
Add color and border to dialog box.

maxtance committed:
Add Intro and Quick Start to README.md

maxtance committed:
Add features for README.md

maxtance committed:
Include links to features

maxtance committed:
Add additional instructions

maxtance committed:
Fix bug to build jar file correctly

maxtance committed:
Merge commit '2396c1dfb4b36100280900770b1a31a87f55500c'

maxtance committed:
Update README.md

maxtance committed:
Update README.md
